src/export/StructureDefinitionExporter.ts

```diff
@@ -414,14 +414,38 @@ export class StructureDefinitionExporter implements Fishable {
         // The FHIR spec prohibits constraining any parent element in a 'specialization'
         // (i.e., logical model and resource), therefore log an error if that is attempted
         // and continue to the next rule.
-        if (element.path !== element.base.path) {
-          // The AddElementRule always sets the newElementBase.path to the value of element.path.
-          // All parent elements will have the element.base.path pointing to the parent.
-          logger.error(
-            `FHIR prohibits logical models and resources from constraining parent elements. Skipping '${rule.constructorName}' at path '${rule.path}' for '${fshDefinition.name}'.`,
-            rule.sourceInfo
-          );
-          continue;
+        const baseElementPath = element.path.replace(element.structDef.id, '').split('.')[1];
+        const parentResource = StructureDefinition.fromJSON(
+          this.fisher.fishForFHIR(fshDefinition.parent)
+        );
+        const inheritedElement = parentResource.findElement(
+          `${fshDefinition.parent}.${baseElementPath}`
+        );
+        if (inheritedElement) {
+          // If we're dealing with a Backbone Element, then we'll want to ensure that the sub-element being referenced
+          // was not added on the parent
+          if (
+            inheritedElement.type.some(typeObj => typeObj.getActualCode() === 'BackboneElement')
+          ) {
+            const parentElementPath = element.path.replace(
+              element.structDef.id,
+              fshDefinition.parent
+            );
+            const parentElement = parentResource.findElement(parentElementPath);
+            if (parentElement) {
+              logger.error(
+                `FHIR prohibits logical models and resources from constraining parent elements. Skipping '${rule.constructorName}' at path '${rule.path}' for '${fshDefinition.name}'.`,
+                rule.sourceInfo
+              );
+              continue;
+            }
```

Comment:
Based on the coverage report, there don't appear to be any test cases that go into this block. Could you add tests for `BackboneElement` elements? There should probably be a test for one that is newly-defined and one that is inherited.

Reply:
Commit `b765ed5` changes the approach to only use a single if-statement, so the existing tests should be alright now!